### PR TITLE
Cypress/FE-2148 switch and select validations components

### PR DESCRIPTION
**Pull request by KatarzynaQA**
# Description
Added tests for classic and default select and switch validations stories

### CODE REVIEW

File: cypress/support/step-definitions/detail-steps.js [hunk 1]
```diff
@@ -20,3 +21,7 @@ Then('icon not exists on preview', () => {
 Then('icon on preview is {string}', (iconName) => {
   icon().should('have.attr', 'data-element', iconName);
 });
+
+Then('icon name on preview is {string}', (iconName) => {
```
Comment (DlgSHi) [inline cypress/support/step-definitions/detail-steps.js]: we need to inform that this function is using only for iFrame rendered page

File: cypress/features/regression/validations/selectValidations.feature [hunk 2]
```diff
@@ -5,14 +5,14 @@ Feature: Select validations component
...
+      And I hover mouse onto icon into iFrame
+    Then tooltipPreview on preview into iFrame is set to '<text>'
+      And icon name on preview is "<state>"
+    Examples:
       | state    | keyWord | text                                 |
       | info     | Brown   | You have selected "Brown"            |
-      | warning  | Blue    | Selecting "Blue" is not recommended  |
-      | error    | Black   | "Black" cannot be selected!          |
\ No newline at end of file
+      | warning  | Blue    | Selecting "Blue" is not recommended  |
```
Comment (DlgSHi) [inline cypress/features/regression/validations/selectValidations.feature]: where is the `error` state?

File: cypress/support/step-definitions/date-input-steps.js [hunk 3]
```diff
@@ -79,12 +80,20 @@ When('I type specific date {string}', (day) => {
...
+
 Then('the date is set to today', () => {
   dayPickerDay(TODAY_CALENDAR).should('have.attr', 'aria-label').should('contains', TODAY_CALENDAR);
   dayPickerDay(TODAY_CALENDAR).should('have.attr', 'class').should('contains', `${DAY_PICKER_PREFIX}today`);
   dayPickerDay(TODAY_CALENDAR).should('have.attr', 'class').should('contains', `${DAY_PICKER_PREFIX}selected`);
 });
 
-And('I click onto specific day {string} via DayPicker for validation component', (specificDay) => {
+Then('I click onto specific day {string} via DayPicker for validation component', (specificDay) => {
   dayPickerDay(`${specificDay}`).click();
```
Comment (DlgSHi) [inline cypress/support/step-definitions/date-input-steps.js]: should be like this -> `dayPickerDay(specificDay).click();`

File: cypress/support/step-definitions/date-input-steps.js [hunk 4]
```diff
@@ -79,12 +80,20 @@ When('I type specific date {string}', (day) => {
...
-})
+});
+
+Then('I click onto specific day {string} via DayPicker for validation component into iFrame', (specificDay) => {
+  dayPickerDayNoIframe(`${specificDay}`).click();
```
Comment (DlgSHi) [inline cypress/support/step-definitions/date-input-steps.js]: should be like this ->  dayPickerDayNoIframe(specificDay).click();

File: cypress/support/step-definitions/switch-steps.js [hunk 5]
```diff
@@ -40,3 +40,11 @@ Then('Switch component is not loading', () => {
   switchInput().should('not.have.attr', 'disabled');
   switchInput().should('not.have.attr', 'readonly');
 });
+
+Then('I click onto {int} switch', (position) => {
+  switchDataComponent(position).click();
+});
+
+Then('I click onto {int} switch again', (position) => {
```
Comment (DlgSHi) [inline cypress/support/step-definitions/switch-steps.js]: we don't need to copy/paste same code, just make it like -> 
`When('I hit Tab key {int} time(s)', (times) => {
  pressTABKey(times);
});`

File: cypress/support/step-definitions/common-steps.js [hunk 6]
```diff
@@ -125,10 +126,23 @@ When('I hover mouse onto icon', () => {
...
 });
 
+When('I hover mouse onto icon into iFrame', () => {
+  cy.wait(100, { log: DEBUG_FLAG }); // required becasue element might be reloaded
+  iconNoIframe().trigger('mouseover');
+});
+
+Then('I hover mouse onto {string} icon', (name) => {
+  getElementNoIframe(name).trigger('mouseover');
+});
```
Comment (DlgSHi) [inline cypress/support/step-definitions/common-steps.js]: probably we could use one of this functions above

File: cypress/support/step-definitions/detail-steps.js (hunk 1 above)
Comment (KatarzynaQA) [reply cypress/support/step-definitions/detail-steps.js]: fixed

File: cypress/features/regression/validations/selectValidations.feature (hunk 2 above)
Comment (KatarzynaQA) [reply cypress/features/regression/validations/selectValidations.feature]: fixed

File: cypress/support/step-definitions/date-input-steps.js (hunk 3 above)
Comment (KatarzynaQA) [reply cypress/support/step-definitions/date-input-steps.js]: fixed

File: cypress/support/step-definitions/common-steps.js (hunk 6 above)
Comment (KatarzynaQA) [reply cypress/support/step-definitions/common-steps.js]: fixed

File: cypress/support/step-definitions/switch-steps.js (hunk 5 above)
Comment (KatarzynaQA) [reply cypress/support/step-definitions/switch-steps.js]: fixed

**Pull request merged**